currency/code_test.go

```diff
@@ -147,25 +155,38 @@ func TestBaseCode(t *testing.T) {
 			main.HasData())
 	}
 
-	catsCode := main.Register("CATS")
+	_ = main.Register("CATS", Unset)
+	if !main.HasData() {
+		t.Errorf("BaseCode HasData() error expected true but received %v",
+			main.HasData())
+	}
+
+	// Changes unset to fiat
+	catsCode := main.Register("CATS", Fiat)
+	if !main.HasData() {
+		t.Errorf("BaseCode HasData() error expected true but received %v",
+			main.HasData())
+	}
+
+	_ = main.Register("CATS", Stable)
 	if !main.HasData() {
 		t.Errorf("BaseCode HasData() error expected true but received %v",
 			main.HasData())
 	}
 
-	if !main.Register("CATS").Match(catsCode) {
+	if !main.Register("CATS", Unset).Equal(catsCode) {
```

Comment:
Registering as unset like with currency.NewCode(); will return the first match. So this is still true. The only caveat is if we call that function with a set role like Stable, then it will not equal.

Reply:
I understand, as I explained, but its not testing with enough depth here. Yes, you return a fiat match, but look at how the test appears, its saying that Unset == Fiat which is not true. I'm saying please update the test to return the currency of main.Register as a variable, then check that the Role of the variable is Fiat in order to highlight what's going on underneath